server/schedule/placement/fit.go

```diff
@@ -115,48 +114,134 @@ func compareRuleFit(a, b *RuleFit) int {
 	}
 }
 
+// StoreSet represents the store container.
+type StoreSet interface {
+	GetStores() []*core.StoreInfo
+	GetStore(id uint64) *core.StoreInfo
+}
+
 // FitRegion tries to fit peers of a region to the rules.
-func FitRegion(stores core.StoreSetInformer, region *core.RegionInfo, rules []*Rule) *RegionFit {
-	peers := prepareFitPeers(stores, region)
+func FitRegion(stores StoreSet, region *core.RegionInfo, rules []*Rule) *RegionFit {
+	w := newFitWorker(stores, region, rules)
+	w.run()
+	return &w.bestFit
+}
```

Comment:
I suggest adding unit test for `FitRegion` so that we can add test cases to test `FitRegion` and compare regionFit result as a whole system.

Reply:
The unit test does use `FitRegion` (but the test name is not correct now, I'll update it)